File: airesources/Rust/src/hlt/logging.rs
use std::fs::File;
use std::io::Write;

pub struct Logger(File);

impl Logger {
    pub fn new(user_id: usize) -> Logger {
        let file = File::create(format!("log_{}.txt", user_id)).expect("Couldn't open file for logging!");
        Logger(file)
    }

    pub fn log(&mut self, message: &str) {
        self.0.write(message.as_bytes()).expect("Couldn't write to log!");
        self.0.write("\n".as_bytes()).expect("Couldn't write to log!");
    }
}

File: airesources/Julia/hlt/logging.jl
using Memento

function set_logger(game::Game)
    logger = getlogger(current_module())            
    logger.handlers = Dict{Any,Memento.Handler}()   # Remove console handler
    push!(logger, DefaultHandler(string(game.botname, "_", game.id, ".log")))

    setlevel!(logger, "info")

    for i in values(logger.handlers)
        i.fmt = DefaultFormatter("{msg}")   # Rewrite logging message format
    end

    return logger
end
